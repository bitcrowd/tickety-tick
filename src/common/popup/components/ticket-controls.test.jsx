import { shallow } from 'enzyme';
import React from 'react';

import { ticket as make } from '../../../../test/factories';
import TicketControls from './ticket-controls';

describe('ticket-controls', () => {
  function render(overrides) {
    const defaults = {
      tickets: ['raz', 'dva', 'tri'].map((title, i) =>
        make({ id: `${i + 1}`, title })
      ),
    };
    const props = { ...defaults, ...overrides };
    return shallow(<TicketControls {...props} />);
  }

  it('renders', () => {
    expect(() => render({})).not.toThrow();
  });
});
