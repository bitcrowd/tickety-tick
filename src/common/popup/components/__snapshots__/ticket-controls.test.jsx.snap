// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ticket-controls renders for a single ticket 1`] = `
<form>
  <div
    className="d-flex flex-row mb-2 btn-group btn-group-lg"
    role="group"
  >
    <TicketCopyButton
      icon={[Function]}
      label="Branch"
      tabIndex={1}
      title="Copy branch name"
      value="branch-1"
    />
    <TicketCopyButton
      icon={[Function]}
      label="Message"
      tabIndex={1}
      title="Copy commit message"
      value="commit-1"
    />
    <TicketCopyButton
      icon={[Function]}
      label="Command"
      tabIndex={1}
      title="Copy CLI command"
      value="command-1"
    />
  </div>
  <div>
    <select
      className="form-control form-control-sm"
      disabled={true}
      onChange={[Function]}
      tabIndex={1}
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        raz
      </option>
    </select>
  </div>
</form>
`;

exports[`ticket-controls renders for multiple tickets 1`] = `
<form>
  <div
    className="d-flex flex-row mb-2 btn-group btn-group-lg"
    role="group"
  >
    <TicketCopyButton
      icon={[Function]}
      label="Branch"
      tabIndex={1}
      title="Copy branch name"
      value="branch-1"
    />
    <TicketCopyButton
      icon={[Function]}
      label="Message"
      tabIndex={1}
      title="Copy commit message"
      value="commit-1"
    />
    <TicketCopyButton
      icon={[Function]}
      label="Command"
      tabIndex={1}
      title="Copy CLI command"
      value="command-1"
    />
  </div>
  <div>
    <select
      className="form-control form-control-sm"
      disabled={false}
      onChange={[Function]}
      tabIndex={1}
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        raz
      </option>
      <option
        key="1"
        value={1}
      >
        dva
      </option>
      <option
        key="2"
        value={2}
      >
        tri
      </option>
    </select>
  </div>
</form>
`;
